tests/TestTotp.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (C) 2017 Weslly Honorato <﻿weslly@protonmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "TestTotp.h"
+
+#include <QTest>
+#include <QTime>
+#include <QDateTime>
+#include <QtEndian>
+#include <QTextCodec>
+
+#include "crypto/Crypto.h"
+#include "totp/totp.h"
+
+QTEST_GUILESS_MAIN(TestTotp)
+
+void TestTotp::initTestCase()
+{
+    QVERIFY(Crypto::init());
+}
+
+
+void TestTotp::testSecret()
```

Comment:
@weslly maybe name it `testParseSecret`, since it's testing the `QTotp::parseOtpString` function.